tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleModuleProjectImporter.java

```diff
@@ -37,49 +37,67 @@ public IStatus canImport( String location )
 
         File file = new File( location );
 
-        if( file.exists() )
+        if( findGradleFile( file ) && findSettingsFile( file ) )
+        {
+            retval = Status.OK_STATUS;
+        }
+        else
         {
-            if( findGradleFile( file ) )
-            {
-                retval = Status.OK_STATUS;
-            }
-
             File parent = file.getParentFile();
 
             while( parent != null )
             {
-                if( findGradleFile( parent ) )
+                if( findGradleFile( parent ) && findSettingsFile( file ) )
                 {
```

Comment:
So with your previous code, we couldn't import the "blade/liferay-gradle" projects, it said that the "liferay-gradle" project was not the root project, because it was looking one directory higher and finding build.gradle in the parent directory.  But liferay-gradle is the root, because it has both a "build.gradle" and a settings.gradle, that makes it a "root" project according to this:  https://docs.gradle.org/current/userguide/intro_multi_project_builds.html

```
8.1. Structure of a multi-project build
Such builds come in all shapes and sizes, but they do have some common characteristics:

A settings.gradle file in the root or master directory of the project
A build.gradle file in the root or master directory
Child directories that have their own *.gradle build files (some multi-project builds may omit child project build scripts)
```